game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProOddsCalculator.java

```diff
@@ -85,18 +85,23 @@ public ProBattleResult estimateDefendBattleResults(final Territory t,
     return callBattleCalculator(t, attackingUnits, defendingUnits, bombardingUnits);
   }
 
-  public ProBattleResult calculateBattleResults(final Territory t,
-      final List<Unit> attackingUnits, final List<Unit> defendingUnits, final Set<Unit> bombardingUnits) {
+  public ProBattleResult calculateBattleResults(final Territory t, final List<Unit> attackingUnits,
+      final List<Unit> defendingUnits, final Set<Unit> bombardingUnits) {
+    return calculateBattleResults(t, attackingUnits, defendingUnits, bombardingUnits, true);
+  }
+
+  public ProBattleResult calculateBattleResults(final Territory t, final List<Unit> attackingUnits,
```

Comment:
Nitpicking the API, any thoughts to removing the boolean parameter?

Looking at this diff, it looks like if `calculateBattleResults` with the boolean parameter were made private, the boolean parameter could be made class-internal by adding something like:

```
  public ProBattleResult calculateBattleResultsNoSubmerge(final Territory t, final List<Unit> attackingUnits,
      final List<Unit> attackingUnits, final List<Unit> defendingUnits, final Set<Unit> bombardingUnits) {
      final List<Unit> defendingUnits, final Set<Unit> bombardingUnits) {
    return calculateBattleResults(t, attackingUnits, defendingUnits, bombardingUnits, false);
  }
```

Reply:
I think you can argue it either way in this case given that the method is only used in one place. But I went ahead and added a method to avoid exposing the boolean.

Reply:
From my perspective it is the class boundary that makes the difference. Regardless thanks for the update.